### PR TITLE
Remove SharedCache::getNumericStorageId to let CacheWrapper do it

### PR DESCRIPTION
**Pull request by MorrisJobke**
* downstream https://github.com/owncloud/core/pull/27172

cc @icewind1991 does this make sense?

### CODE REVIEW

File: apps/files_sharing/lib/Cache.php [hunk 1]
```diff
@@ -71,14 +71,6 @@ public function getCache() {
 		return $this->cache;
 	}
 
-	public function getNumericStorageId() {
```
Comment (icewind1991) [inline apps/files_sharing/lib/Cache.php]: Removing this breaks the sharing performance improvements

**Review by icewind1991 — DISMISSED** [review]
👎 For removing it from the shared cache

Having the tests from the rest of the commit would be nice though

**Review by icewind1991 — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Unit tests fail:

```
1) OCA\Files_Sharing\Tests\CacheTest::testNumericStorageId
Failed asserting that false matches expected 344.
/drone/src/github.com/nextcloud/server/apps/files_sharing/tests/CacheTest.php:550
```

**Comment by icewind1991** [comment]
Fixed

**Review by rullzer — APPROVED** [review]

**Pull request merged**